src/canmatrix/tests/test_canmatrix.py

```diff
@@ -355,6 +355,64 @@ def test_encode_decode_frame():
     assert decoded_data == input_data
 
 
+# Define tests
+def test_define_set_default():
+    define = canmatrix.canmatrix.Define("")
+    define.setDefault("string")
+    assert define.defaultValue == "string"
+    define.setDefault('"quoted_string"')
+    assert define.defaultValue == "quoted_string"
+
+
+def test_define_update_enum_definition():
+    define = canmatrix.canmatrix.Define("")
+    define.type = "ENUM"
+    define.values = ["ready", "off"]
+    define.update()
+    assert define.definition == 'ENUM "ready","off"'
+
+
+def test_define_update_ingored_non_enum():
+    def_str = "INT 0 100"
+    define = canmatrix.canmatrix.Define(def_str)
+    define.update()
+    assert  define.definition == def_str
+
+
+def test_define_for_int():
+    define = canmatrix.canmatrix.Define("INT -5 10")
+    assert define.type == "INT"
+    assert define.min == -5
+    assert define.max == 10
+
+
+# def test_define_for_hex():
+#     define = canmatrix.canmatrix.Define("HEX ? ?")
+#     assert define.type == "HEX"
+#     assert define.min == ?
+#     assert define.max == ?
+
+
+def test_define_for_string():
+    define = canmatrix.canmatrix.Define("STRING")
+    assert define.type == "STRING"
+    assert define.min is None
+    assert define.max is None
+
+
+def test_define_for_enum():
+    define = canmatrix.canmatrix.Define('ENUM "red","green", blue')
```

Comment:
Red and green are quoted but blue is not.  If the quotes provide any meaning let's exercise it.  Can enumeration names have spaces?  Special characters?  Comma inside quotes?

Reply:
The missing quotes and additional space are by intent to reflect the code in the function under the test. That means it again tests multiple things: quote removal, trim() used, all values parsed.
But it only reflects the code, I have no additional knowledge about it.

Reply:
You meant to split it to multiple tests, right?

Reply:
Maybe multiple tests, but my point was to exercise some corner cases.  You obviously already covered some of them but I was suggesting more.  I'd either ~~comment~~ docstring (uh, hmm...) or make separate tests.  If the present code can handle something like `'ENUM "red", "green, blue, and cyan", magenta'` than let's have a test for the internal `,`s and ` `s inside the enumeration value.

Reply:
not supported so withdrawing suggestion